packages/core/test/publisher.spec.ts

```diff
@@ -1,36 +1,33 @@
-import { expect, describe, it, beforeEach } from "vitest";
+import { expect, describe, it, beforeEach, afterEach } from "vitest";
 import pino from "pino";
 import Sinon from "sinon";
 import { getDefaultLoggerOptions } from "@walletconnect/logger";
-import { IRelayer } from "@walletconnect/types";
+import { ICore, IPublisher, IRelayer } from "@walletconnect/types";
 import { generateRandomBytes32, hashMessage } from "@walletconnect/utils";
 import { Publisher } from "../src/controllers/publisher";
 import { HEARTBEAT_EVENTS } from "@walletconnect/heartbeat";
 
 import { Core, CORE_DEFAULT, PUBLISHER_DEFAULT_TTL, Relayer } from "../src";
-import { TEST_CORE_OPTIONS } from "./shared";
+import { disconnectSocket, TEST_CORE_OPTIONS, throttle } from "./shared";
 
 describe("Publisher", () => {
   const logger = pino(getDefaultLoggerOptions({ level: CORE_DEFAULT.logger }));
 
-  let relayer: IRelayer;
+  let core: ICore;
   let publisher: Publisher;
 
   beforeEach(async () => {
-    const core = new Core(TEST_CORE_OPTIONS);
+    if (core) {
+      await disconnectSocket(core);
```

Comment:
why share the core with each test?

Reply:
its not shared, its reassigned a few lines later. Just to make sure on the next iteration the sockets are disconnected